mscore/workspace.cpp

```diff
@@ -76,26 +76,41 @@ static QString editedWorkspaceTranslatableName(const QString& oldWorkspaceTransl
 
 static QString defaultWorkspaceTranslatableName(const QString& editedWorkspaceName)
       {
-      const auto it = std::find(defaultEditedWorkpaces.begin(), defaultEditedWorkpaces.end(), editedWorkspaceName);
+      const auto it = std::find(WorkspacesManager::defaultEditedWorkspaces.begin(), WorkspacesManager::defaultEditedWorkspaces.end(), editedWorkspaceName);
 
-      if (it != defaultEditedWorkpaces.end()) {
-            const int idx = it - defaultEditedWorkpaces.begin();
-            if (idx < int(defaultWorkspaces.size()))
-                  return defaultWorkspaces[idx];
+      if (it != WorkspacesManager::defaultEditedWorkspaces.end()) {
+            const int idx = it - WorkspacesManager::defaultEditedWorkspaces.begin();
+            if (idx < int(WorkspacesManager::defaultWorkspaces.size()))
+                  return WorkspacesManager::defaultWorkspaces[idx];
             }
 
       return QString();
       }
 
-//---------------------------------------------------------
-//   undoWorkspace
-//---------------------------------------------------------
-
-void MuseScore::undoWorkspace()
+/**
+ *   Reverts current workspace to the state of the `source` workspace
+ */
+void MuseScore::resetWorkspace()
       {
-      // TODO: make a separate session start backup?
-      WorkspacesManager::currentWorkspace()->read();
-      WorkspacesManager::currentWorkspace()->setDirty(false);
+      //if currentWorkspace is the `edited` Basic or Advanced one, remove the edited and show the source one
+      if (WorkspacesManager::isDefaultEditedWorkspace(WorkspacesManager::currentWorkspace())) {
+            //Basic edited
+            const QString& currWorkspaceName = WorkspacesManager::currentWorkspace()->translatableName();
+            if (currWorkspaceName == WorkspacesManager::defaultEditedWorkspaces[0]) {
+                  WorkspacesManager::remove(WorkspacesManager::currentWorkspace());
+                  changeWorkspace(WorkspacesManager::findByTranslatableName(WorkspacesManager::defaultWorkspaces[0]));
+                  }
+            //Advanced edited
+            else if (currWorkspaceName == WorkspacesManager::defaultEditedWorkspaces[1]) {
+                  WorkspacesManager::remove(WorkspacesManager::currentWorkspace());
+                  changeWorkspace(WorkspacesManager::findByTranslatableName(WorkspacesManager::defaultWorkspaces[1]));
```

Comment:
I get a crash here (well, further down the line, when changeWorkspace tries to save the current workspace which has just been removed).  I was able to fix it by changing this code to do the same as removeWorkspace: first call Workspaces::Manager::setCurrentWorkspace on this workspace, *then* do the change

Reply:
Also, do we *really* know workspace 0 is always Basic and 1 is always Advanced?  Maybe we do, but as mentioned elsewhere, I've seen cases where combobox ordering is different from expected, even when choosing Basic in startup wizard.